datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1935,6 +1942,73 @@ impl Filter {
 
         Ok(Self { predicate, input })
     }
+
+    /// Is this filter guaranteed to return 0 or 1 row in a given instantiation?
+    ///
+    /// This function will return `true` if its predicate contains a conjunction of
+    /// `col(a) = <expr>`, where its schema has a unique filter that is covered
+    /// by this conjunction.
+    ///
+    /// For example, for the table:
+    /// ```sql
+    /// CREATE TABLE t (a INTEGER PRIMARY KEY, b INTEGER);
+    /// ```
+    /// `Filter(a = 2).is_scalar() == true`
+    /// , whereas
+    /// `Filter(b = 2).is_scalar() == false`
+    /// and
+    /// `Filter(a = 2 OR b = 2).is_scalar() == false`
+    fn is_scalar(&self) -> bool {
```

Comment:
I suggest use `Uniform slot` and `Unique slot` to describe `FDs` instead of `scalar`.

Commonly used Functional dependencies : including

`uniform slot`, which means `ndv <= 1`
`unique slot`, which means `ndv = row`

Reply:
Do I understand correctly that the suggestion is to rename this method to `Filter::is_uniform()`?

Reply:
Yes, and add these definition into doc/